src/Socket.js

```diff
@@ -54,7 +54,10 @@ export class Socket extends EventEmitter {
         this.connected = false
         this.emit('disconnected')
         if (this.opts.reconnect) {
-          this.reconnect()
+          this.reconnect().catch(() => {
+            let err = new Error(`Auto-reconnect failed after ${this.maxRetries} retries`)
```

Comment:
I asked myself the same question.
I added a new one because `Too many connection failures` doesn't talk about "reconnection". But I guess it would make more sens to just rephrase it.
Let me change this.